### PR TITLE
pasco: init at 20040505_1

### PR DESCRIPTION
**Pull request by D3vil0p3r**
## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by D3vil0p3r** [comment]
@a-n-n-a-l-e-e I used here `"CC=cc"` but still get the error on Darwin:
```
gcc: command not found
```

**Comment by a-n-n-a-l-e-e** [comment]
> @a-n-n-a-l-e-e I used here `"CC=cc"` but still get the error on Darwin:
> 
> ```
> gcc: command not found
> ```

so the makefile doesn't use the usuall `CC` variable so we can't override it via the command line here.


```make
all: install

install:  pasco.c
        gcc -o pasco pasco.c -lm -lc;cp pasco ../bin

installwin:  pasco.c
        gcc -DCYGWIN -o pasco.exe pasco.c -lm -lc;cp pasco.exe ../bin

clean: 
        rm -f pasco.exe pasco
```

**Comment by D3vil0p3r** [comment]
> > @a-n-n-a-l-e-e I used here `"CC=cc"` but still get the error on Darwin:
> > ```
> > gcc: command not found
> > ```
> 
> so the makefile doesn't use the usuall `CC` variable so we can't override it via the command line here.
> 
> ```makefile
> all: install
> 
> install:  pasco.c
>         gcc -o pasco pasco.c -lm -lc;cp pasco ../bin
> 
> installwin:  pasco.c
>         gcc -DCYGWIN -o pasco.exe pasco.c -lm -lc;cp pasco.exe ../bin
> 
> clean: 
>         rm -f pasco.exe pasco
> ```

Is it enough if we replace `gcc` to `cc` by `substituteInPlace`?

**Comment by a-n-n-a-l-e-e** [comment]
> Is it enough if we replace `gcc` to `cc` by `substituteInPlace`?

sure

File: pkgs/by-name/pa/pasco/package.nix [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "Examine the contents of Internet Explorer's cache files for forensic purposes";
+    mainProgram = "pasco";
+    homepage = "https://sourceforge.net/projects/fast/files/Pasco/";
+    maintainers = with maintainers; [ d3vil0p3r ];
+    platforms = platforms.unix;
+    license = with licenses; [ free ];
```
Comment (a-n-n-a-l-e-e) [inline pkgs/by-name/pa/pasco/package.nix:41]:
```suggestion
    license = with licenses; [ free ];
```
https://salsa.debian.org/pkg-security-team/pasco/-/blob/debian/master/src/pasco.c?ref_type=heads
think this is a bsd3 -- looks like: https://opensource.org/license/BSD-3-clause/

File: pkgs/by-name/pa/pasco/package.nix [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  pname = "pasco";
+  version = "20040505_1";
+
+  src = fetchurl {
+    url = "mirror://sourceforge/project/fast/Pasco/Pasco%20v${finalAttrs.version}/pasco_${finalAttrs.version}.tar.gz";
+    hash = "sha256-o7jue+lgVxQQvFZOzJMGd1WihlD7Nb+1WaSutq9vaGg=";
+  };
+
+  patches = [
+    ./pasco.patch
```
Comment (a-n-n-a-l-e-e) [inline pkgs/by-name/pa/pasco/package.nix:16]:
```suggestion
    ./include-string.h.patch
```
patches need a comment or should be named well to not need a comment.

**Comment by D3vil0p3r** [comment]
> > Is it enough if we replace `gcc` to `cc` by `substituteInPlace`?
> 
> sure

Fixed

File: pkgs/by-name/pa/pasco/package.nix (hunk 1 above)
Comment (D3vil0p3r) [reply pkgs/by-name/pa/pasco/package.nix]: Thank you. Is there a reason when I save the package.nix code by VSCodium and I save and push the change, the EditorConfig check produces an error related `Wrong new line` at the end despite I didn't put any new line at the end of the file?

**Review by ghost — APPROVED** [review]

File: pkgs/by-name/pa/pasco/package.nix (hunk 1 above)
Comment (a-n-n-a-l-e-e) [reply pkgs/by-name/pa/pasco/package.nix]: https://github.com/NixOS/nixpkgs/blob/master/.editorconfig says that the last line of the file must end with a new line -- maybe your editor isn't adding them? https://stackoverflow.com/questions/44704968/visual-studio-code-insert-newline-at-the-end-of-files

**Pull request merged**